### ISSUE TITLE
Can't publish using newer version of exp

### ISSUE THREAD

**Issue by fvieira** [expo/expo#1332]
Lately I've been unable to publish using exp because it says it cannot find package.json.
The last version that I managed to publish with was 47.0.0.
Here's the command and the full output from it:

```
$ exp --version
49.0.1
$ EXPO_DEBUG=true exp publish -c prod
[exp] Unable to find an existing exp instance for this directory, starting a new one...
[exp] Error: Can't find package.json
[exp] Warning: You are using npm version 5.6.0. There may be bugs in this version, use it at your own risk. We recommend version 4.6.1.
[exp] Couldn't start project. Please fix the errors and restart the project.
[exp] Error: Couldn't start project. Please fix the errors and restart the project.
    at /xdl/src/Project.js:1317:11
    at Generator.next (<anonymous>)
    at step (/home/fvieira/.nvm/versions/node/v8.9.3/lib/node_modules/exp/node_modules/xdl/build/Project.js:1710:191)
    at /home/fvieira/.nvm/versions/node/v8.9.3/lib/node_modules/exp/node_modules/xdl/build/Project.js:1710:361
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

```

I know there's a warning there, but I tried downgrading npm and it didn't fix the problem.
Weirdly enough, if I do a build command, it publishes with no problems...

**Comment by raarts** [comment]
Having the exact same problem here.

**Comment by fson** [comment]
Hi, which `exp` version are you using? Does the issue still exist if you upgrade to exp 49.1.0?

**Comment by fvieira** [comment]
Yup, no difference...

```
$ exp --version
49.1.0
$ env EXPO_DEBUG=true exp publish -c prod
[exp] Unable to find an existing exp instance for this directory, starting a new one...
[exp] Error: Can't find package.json
[exp] Warning: You are using npm version 5.6.0. There may be bugs in this version, use it at your own risk. We recommend version 4.6.1.
[exp] Couldn't start project. Please fix the errors and restart the project.
[exp] Error: Couldn't start project. Please fix the errors and restart the project.
    at /xdl/src/Project.js:1319:11
    at Generator.next (<anonymous>)
    at step (/home/fvieira/.nvm/versions/node/v8.9.3/lib/node_modules/exp/node_modules/xdl/build/Project.js:1717:191)
    at /home/fvieira/.nvm/versions/node/v8.9.3/lib/node_modules/exp/node_modules/xdl/build/Project.js:1717:361
    at <anonymous>
```

**Comment by fson** [comment]
Does a file named `package.json` exist in the root of your project folder? And could you share the contents of your `app.json` file?

**Comment by fvieira** [comment]
Yes, that `package.json` file is there, I can publish without problems using version 47.0.0.

Here's my `app.json` file, minus some redacted values:
```
{
  "expo": {
    "name": "Entrega em Casa",
    "slug": "REDACTED",
    "privacy": "unlisted",
    "sdkVersion": "25.0.0",
    "version": "1.00.07",
    "orientation": "portrait",
    "primaryColor": "#242729",
    "icon": "./assets/logo-app.android.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "cover"
    },
    "ios": {
      "buildNumber": "1.00.07",
      "bundleIdentifier": "pt.entregaemcasa.clientapp",
      "icon": "./assets/logo-app.ios.png",
      "infoPlist": {
        "NSLocationWhenInUseUsageDescription": "Para conseguirmos entregar o seu pedido à sua porta"
      }
    },
    "android": {
      "versionCode": 10007,
      "package": "pt.entregaemcasa.clientapp",
      "icon": "./assets/logo-app.android.png",
      "permissions": ["ACCESS_FINE_LOCATION"],
      "config": {
        "googleMaps": {
          "apiKey": "REDACTED"
        }
      }
    },
    "hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": "francisco-vieira",
            "project": "entrega-em-casa",
            "authToken": "REDACTED"
          }
        }
      ]
    }
  }
}
```

**Comment by fvieira** [comment]
Ok, good news, found the bug!

After a bit of investigation, found the problem only occurred when I was using a release channel, more specifically, if I was using the `-c` flag. It all worked if I used `--release-channel`.
After that, checking the first commit after the 47.0.0 release revealed an obvious problem:

```
+    .option('-c, --clear', 'Clear the React Native packager cache')
     .option('-c --release-channel <release channel>',
     ...
```

A new flag was added that clashes with the channel one...

Here's the commit for reference:
https://github.com/expo/exp/commit/de62de5974cea3b8237cc5fc94d0f6e9ef395fc3

I won't open a PR just because I don't know what flag you want to keep, but the fix is simple enough...
